File: app/src/main/java/com/example/AppComponent.java
package com.example;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {
  AndroidInjectionModule.class,
  AppModule.class
})
public interface AppComponent {
  void inject(App application);

  @Component.Builder
  interface Builder {
    @BindsInstance
    Builder application(Application application);

    AppComponent build();
  }
}

File: app/src/main/kotlin/br/com/vitorsalgado/example/AppComponent.kt
package br.com.vitorsalgado.example

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [(AndroidInjectionModule::class), (AppModule::class)])
interface AppComponent {
  fun inject(application: App)

  @Component.Builder
  interface Builder {
    @BindsInstance
    fun application(application: Application): Builder

    fun build(): AppComponent
  }
}
